plugins/woocommerce/src/Database/Migrations/CustomOrderTable/WPPostToCOTMigrator.php

```diff
@@ -0,0 +1,540 @@
+		$result  = $wpdb->query( $queries ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- $queries is already prepared.
+		if ( count( $order_data['data'] ) !== $result ) {
+			// Some rows were not inserted.
+			// TODO: Find and log the entity ids that were not inserted.
+			echo ' error ';
+		}
+
+		$order_post_ids = array_column( $order_data['data'], 'post_id' );
+		$this->process_next_migrator_batch( $this->billing_address_table_migrator, $order_post_ids, $order_by );
+		$this->process_next_migrator_batch( $this->shipping_address_table_migrator, $order_post_ids, $order_by );
+		$this->process_next_migrator_batch( $this->operation_data_table_migrator, $order_post_ids, $order_by );
+
+		$last_post_migrated = max( array_keys( $order_data['data'] ) );
+		$this->update_checkpoint( $last_post_migrated );
+
+		return false;
+	}
+
+	/**
+	 * Process next batch for a given address type.
+	 *
+	 * @param MetaToCustomTableMigrator $migrator Migrator instance for address type.
+	 * @param array                     $order_post_ids Array of post IDs for orders.
+	 * @param string                    $order_by Order by clause.
+	 */
+	private function process_next_migrator_batch( $migrator, $order_post_ids, $order_by ) {
+		global $wpdb;
+		$post_ids_where_clause = $this->get_where_id_clause( $order_post_ids, 'post_id' );
+		$batch_size            = count( $order_post_ids );
+		$data                  = $migrator->fetch_data_for_migration( $post_ids_where_clause, $batch_size, $order_by );
+		foreach ( $data['errors'] as $order_id => $error ) {
+			// TODO: Add name of the migrator in error message.
+			$this->error_logger->log( 'info', "Error in importing data for Order ID $order_id: " . print_r( $error, true ) );
+		}
+		$queries = $migrator->generate_insert_sql_for_batch( $data['data'], 'insert' );
+		$result  = $wpdb->query( $queries ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- Insert statements should already be escaped.
+		if ( count( $data['data'] ) !== $result ) {
+			// Some rows were not inserted.
+			// TODO: Find and log the entity ids that were not inserted.
+			echo 'error';
+		}
+	}
+
+	/**
+	 * Method to migrate single record.
+	 *
+	 * @param int $post_id Post ID of record to migrate.
+	 *
+	 * @return bool|\WP_Error
+	 */
+	public function process_single( $post_id ) {
+		global $wpdb;
+
+		$where_clause = $wpdb->prepare( 'ID = %d', $post_id );
+		$data         = $this->order_table_migrator->fetch_data_for_migration( $where_clause, 1, 'ID ASC' );
+		if ( isset( $data['errors'][ $post_id ] ) ) {
+			return new \WP_Error( $data['errors'][ $post_id ] );
+		}
+
+		$queries = $this->order_table_migrator->generate_insert_sql_for_batch( $data['data'], 'replace' );
+		$result  = $wpdb->query( $queries ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- $queries is already prepared.
+		if ( 1 !== $result ) {
+			// TODO: Fetch and return last error.
+			echo 'error';
```

Comment:
Given a `WP_Error` is returned below, maybe we can remove this "error" here? Even understanding this is in-progress and only for debugging purposes at this initial state, just seeing "error" being printed doesn't necessarily convey what went wrong.

Reply:
these echos are only placholders for now, we will remove them as we add proper support for surfacing errors.